source/Img32.pas

```diff
@@ -185,8 +185,8 @@   TImage32 = class(TObject)
     function GetIsEmpty: Boolean;
     function GetPixelBase: PColor32;
     function GetPixelRow(row: Integer): PColor32;
-    procedure NearestNeighborResize(newWidth, newHeight: Integer);
-    procedure ResamplerResize(newWidth, newHeight: Integer);
+    procedure NearestNeighborResize(targetImg: TImage32; newWidth, newHeight: Integer);
```

Comment:
Firstly, I'm very aware of the challenges ecountered whenever altering someone else's code, especially making necessary compromises in order to avoid making major design changes. (I am very open to significant redesign suggestions, though for the next few months I'm mosty focusing on another project.)

Having said that, as a very minor design consideration, I'd prefer to move methods in TImage32 that alter external images out that class (which is already unhealthily overweight 😁). So my inclination would be to remove the newly modified `NearestNeighborResize` and `ResamplerResize` out of TImage32 and create or use an existing function in Img32.Resamplers.